src/js/features/ingestion/lastIngestion.store.ts

```diff
@@ -0,0 +1,69 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { katsuLastIngestionsUrl, gohanLastIngestionsUrl } from '@/constants/configConstants';
+import { printAPIError } from '@/utils/error.util';
+
+import { LastIngestionDataTypeResponse, DataTypeMap } from '@/types/lastIngestionDataTypeResponse';
+
+// Async thunks to fetch data from the two endpoints
+export const fetchKatsuData = createAsyncThunk('dataTypes/fetchKatsuData', (_, { rejectWithValue }) =>
+  axios
+    .get(katsuLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+export const fetchGohanData = createAsyncThunk('dataTypes/fetchGohanData', (_, { rejectWithValue }) =>
+  axios
+    .get(gohanLastIngestionsUrl)
+    .then((res) => res.data)
+    .catch(printAPIError(rejectWithValue))
+);
+
+// Define the state structure
+export interface DataTypeState {
+  isFetchingKatsuData: boolean;
+  isFetchingGohanData: boolean;
+  dataTypes: DataTypeMap;
+}
+
+// Initialize the state
+const initialDataTypeState: DataTypeState = {
+  isFetchingKatsuData: false,
+  isFetchingGohanData: false,
+  dataTypes: {},
+};
+
+const reduceServiceDataTypes = (state: DataTypeState, { payload }: PayloadAction<LastIngestionDataTypeResponse[]>) => {
+  state.dataTypes = {
+    ...state.dataTypes,
+    ...Object.fromEntries(payload.map((data) => [data.id, data])),
+  };
+  state.isFetchingKatsuData = false;
+  state.isFetchingGohanData = false;
```

Comment:
this shouldn't set both to false since it is only setting data for one of the data types. you could move the assignment out of this function and make this function just handle the data type assignments, accepting `state` and `payload` as arguments, instead.